File: go/chapter4/example/app/read/linkedread.go
package read

import "fmt"

type Node struct {
	data int
	next *Node
}

func Linkedread() {
	var head *Node = nil
	var tail *Node = nil
	var cur *Node = nil

	var newNode *Node = nil
	var readData int

	for {
		fmt.Printf("자연수 입력: ")
		fmt.Scanf("%d", &readData)
		if readData < 1 {
			break
		}

		newNode = &Node{data: readData, next: nil}

		if head == nil {
			head = newNode
		} else {
			tail.next = newNode
		}
		tail = newNode
	}
	fmt.Printf("\n")

	fmt.Printf("입력 받은 데이터의 전체출력! \n")
	if head == nil {
		fmt.Printf("저장된 자연수가 존재하지 않습니다. \n")
	} else {
		cur = head
		fmt.Printf("%d ", cur.data)

		for cur.next != nil {
			cur = cur.next
			fmt.Printf("%d ", cur.data)
		}
	}
	fmt.Printf("\n\n")

	if head == nil {
		return
	} else {
		cur := head
		for cur != nil {
			fmt.Printf("%d을(를) 삭제합니다. \n", cur.data)
			cur = cur.next
		}
	}
}

File: python/chapter4/example/app/read/linkedread.py
from pydantic import BaseModel
from typing import Optional

class Node(BaseModel):
    data: int
    next: Optional['Node'] = None

def Linkedread():
    head: Node = None
    tail: Node = None
    cur: Node = None

    newNode: Node = None
    readData: int

    while True:
        readData = int(input("자연수 입력: "))
        if readData < 1:
            break

        newNode = Node(
            data=readData,
            next=None
        )

        if head == None:
            head = newNode
        else:
            tail.next = newNode
        
        tail = newNode
    print("\n")

    print("입력 받은 데이터의 전체출력! \n")
    if head == None:
        print("저장된 자연수가 존재하지 않습니다. \n")
    else:
        cur = head
        print(f"{cur.data}", end=" ")

        while cur.next != None:
            cur = cur.next
            print(f"{cur.data}", end=" ")
    
    print("\n\n")

    if head == None:
        return
    else:
        cur = head
        while cur != None:
            print(f"{cur.data}을(를) 삭제합니다. \n")
            cur = cur.next


